modules/gdscript/gdscript_warning.cpp

```diff
@@ -40,10 +40,10 @@ String GDScriptWarning::get_message() const {
 	switch (code) {
 		case UNASSIGNED_VARIABLE:
 			CHECK_SYMBOLS(1);
-			return vformat(R"(The variable "%s" was used before being assigned a value.)", symbols[0]);
+			return vformat(R"(The variable "%s" is being used before being assigned a value.)", symbols[0]);
 		case UNASSIGNED_VARIABLE_OP_ASSIGN:
 			CHECK_SYMBOLS(1);
-			return vformat(R"(Using assignment with operation but the variable "%s" was not previously assigned a value.)", symbols[0]);
+			return vformat(R"(The variable "%s" is being assigned to with an operation but was not previously assigned a value.)", symbols[0]);
```

Comment:
This one still feels a bit hard to parse IMO.
I'm not sure what would better though. For a non-native speaker at least "is being assigned to with" is heavy, especially with the "to with" next to each other.

Might also make sense to say "operator" instead of "operation" which might be clearer to users?

Is there an actual technical terms for operators like `+=` and `*=` which separates them from the `=` assignment operator? Being technically `a = 42` would be assigning something to the variable with an operator too.

Reply:
I agree it's rather confusingly worded. It sounds like operators like `+=` and `*=` are called "compound assignment" or sometimes "augmented assignment" operators. I personally prefer "compound assignment" since it feels a bit more explicit about performing two operations, but it sounds like "augmented assignment" is used for Python.

Reply:
How about:

> The variable "%s" is being mutated with a compound-assignment operator (such as `+=`) but was not previously initialized.

It's a bit technical and still a mouthful, but with the added example and technical name of the operator, it might be easier? I'm assuming that at this stage we don't have easy access to which compound-assignment operator was used? Otherwise including it directly instead of my example would make it much easier to understand.

Reply:
Yeah I hesitated with "modified" too. It's probably more user-friendly / less jargony, so I'm fine with using it instead.